wurst/systems/crafting/CraftTannery.wurst

```diff
@@ -0,0 +1,97 @@
+package CraftTannery
+
+// Extension library imports:
+import ItemRestriction
+
+// Local imports:
+import ItemRecipeExtended
+import LocalObjectIDs
+import GameMode
+
+
+init
+    GameMode.onModeSelectionFinish() ->
+        let tannery = new UnitRequirement("tannery")
+        tannery.addUnit(UNIT_TANNERY)
+
+        let omninousAltar = new UnitRequirement("omninousAltar")
+        omninousAltar.addUnit(UNIT_OMINOUS_ALTAR)
+
+        let elkBoots = new ItemRecipeExt(ITEM_ELK_SKIN_BOOTS, 0, true, true, false, ABILITY_QM_ELK_SKIN_BOOTS)
```

Comment:
I don't really see what you're thinking, I can make another subclass off ItemRecipeExt and hard set the charges, ordered, permanent and pickupable attribute and you'll have a constructor like this, otherwise I don't visualize what you're suggesting.
` let elkBoots = new ItemRecipeArmor(ITEM_ELK_SKIN_BOOTS, ABILITY_QM_ELK_SKIN_BOOTS)`

Reply:
Yeah, something like that. Similar to how object definitions are set incrementally instead of a giant constructor.